### PR TITLE
parameterized Test

### LINKED ISSUE #1371: ConfigsTest: Replace multiple tests with a single parameterized one
### Description

We seem to be testing `Configs.asBoolean` method in ConfigsTest which seems kind of redundant.

https://github.com/eclipse/jkube/blob/b45facd5b3dd25a31850d2f38b96c73edf66b83a/jkube-kit/common/src/test/java/org/eclipse/jkube/kit/common/ConfigsTest.java#L108-L170

[Sonar suggests](https://sonarcloud.io/project/issues?resolved=false&id=jkubeio_jkube&open=AXTUMbfd_bxDuY6CEkWA) replacing these tests with parameterized ones.

- [x] Remove these tests from ConfigsTest
- [x] Create a new parameterized test ConfigsAsBooleanTest which tests all scenarios covered by these tests

### PR DESCRIPTION
**Pull request by baruKreddy**
## Description
Fix #1371

## Type of change

 - [x] Bug fix (non-breaking change which fixes an issue)
 - [ ] Feature (non-breaking change which adds functionality)
 - [ ] Breaking change (fix or feature that would cause existing functionality to change
 - [ ] Chore (non-breaking change which doesn't affect codebase;
   test, version modification, documentation, etc.)

## Checklist
 - [x] I have read the [contributing guidelines](https://www.eclipse.org/jkube/contributing)
 - [x] I signed-off my commit with a user that has signed the [Eclipse Contributor Agreement](https://www.eclipse.org/legal/ECA.php)
 - [ ] I Added [CHANGELOG](../CHANGELOG.md) entry
 - [x] I have implemented unit tests to cover my changes
 - [ ] I have updated the [documentation](../kubernetes-maven-plugin/doc) accordingly
 - [ ] No new bugs, code smells, etc. in [SonarCloud](https://sonarcloud.io/dashboard?id=jkubeio_jkube) report
 - [ ] I tested my code in Kubernetes
 - [ ] I tested my code in OpenShift

### CODE REVIEW

File: jkube-kit/common/src/test/java/org/eclipse/jkube/kit/common/ConfigsAsBooleanTest.java [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+ * which is available at:
+ *
+ *     https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *   Red Hat, Inc. - initial API and implementation
+ */
+package org.eclipse.jkube.kit.common.util;
```
Comment (rohanKanojia) [inline jkube-kit/common/src/test/java/org/eclipse/jkube/kit/common/ConfigsAsBooleanTest.java:14]: Your class's package name is wrong:
```suggestion
package org.eclipse.jkube.kit.common;
```

File: jkube-kit/common/src/test/java/org/eclipse/jkube/kit/common/ConfigsAsBooleanTest.java [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+import java.util.Arrays;
+import java.util.Collection;
+import static org.assertj.core.api.Assertions.assertThat;
+@RunWith(Parameterized.class)
+public class ConfigsAsBooleanTest {
+
+    @Parameterized.Parameters(name = "{index}: asBoolean({0})={1}")
+    public static Collection<Object[]> data() {
+        return Arrays.asList(new Object[][] {
+                { " 1 2 1337", false },{ "1", false },{"0",false},{"fALsE", false },{"false", false}
```
Comment (rohanKanojia) [inline jkube-kit/common/src/test/java/org/eclipse/jkube/kit/common/ConfigsAsBooleanTest.java:30]: This looks good! However, I was thinking if we could also add a description parameter field which might improve readability a little.

```java
    @Parameterized.Parameter
    public String description;
```

Then you can provide this in data array like this (basically porting test names here):
```java
    @Parameterized.Parameters(name = "{0}: asBoolean({1})={2}")
    public static Collection<Object[]> data() {
        return Arrays.asList(new Object[][] {
                { "Unsupported String", " 1 2 1337", false },
                { "One", "1", false },
                { "Zero", "0", false },
                { "FalseMixedCase", "fALsE", false },
                { "False", "false", false},
                // ...           
```

Also, you seem to only have cases with value false here. Please port test cases testing true as well i.e `asBooleanValueWithTrueShouldReturnTrue` `asBooleanValueWithTrueUpperCaseShouldReturnTrue` `asBooleanValueWithTrueMixedCaseShouldReturnTrue`

Comment (baruKreddy) [reply jkube-kit/common/src/test/java/org/eclipse/jkube/kit/common/ConfigsAsBooleanTest.java]: added suggested changes, Thanks!

File: jkube-kit/common/src/test/java/org/eclipse/jkube/kit/common/ConfigsAsBooleanTest.java (hunk 1 above)
Comment (baruKreddy) [reply jkube-kit/common/src/test/java/org/eclipse/jkube/kit/common/ConfigsAsBooleanTest.java]: added changes.

File: jkube-kit/common/src/test/java/org/eclipse/jkube/kit/common/ConfigsAsBooleanTest.java [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+
+import java.util.Arrays;
+import java.util.Collection;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+@RunWith(Parameterized.class)
+public class ConfigsAsBooleanTest {
+
+  @Parameterized.Parameters(name = "{index}: asBoolean({1})={2}")
```
Comment (rohanKanojia) [inline jkube-kit/common/src/test/java/org/eclipse/jkube/kit/common/ConfigsAsBooleanTest.java:28]: You're not adding description parameter here. 
```suggestion
  @Parameterized.Parameters(name = "{index} {0} : asBoolean({1})={2}")
```

**Review by rohanKanojia — APPROVED** [review]

**Review by manusa — APPROVED** [review]

**Pull request merged**